rclc/multithread_publisher_subscriber/main.c

```diff
@@ -0,0 +1,135 @@
+#include <rcl/rcl.h>
+#include <rcl/error_handling.h>
+#include <rclc/rclc.h>
+#include <rclc/executor.h>
+
+#include <std_msgs/msg/header.h>
+
+#include <stdio.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <time.h>
+
+#define STRING_BUFFER_LEN 100
+#define PUBLISHER_NUMBER 3
+
+#define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc); return 1;}}
+#define RCSOFTCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Continuing.\n",__LINE__,(int)temp_rc);}}
+
+struct arg_struct {
+    int index;
+    rcl_publisher_t *publisher;
+};
+
+rcl_node_t node;
+pthread_t pub_thr[PUBLISHER_NUMBER];
+rcl_publisher_t publisher[PUBLISHER_NUMBER];
+rcl_subscription_t subscriber;
+std_msgs__msg__Header recv_msg;
+
+volatile bool exit_flag = false;
+
+// Publish thread
+void* publish(
+        void* args)
+{
+	struct arg_struct *arguments = args;
+	rcl_publisher_t *publisher = arguments->publisher;
+	uint8_t id = arguments->index;
+
+	// Create and allocate the publisher message
+	std_msgs__msg__Header send_msg;
```

Comment:
Init the msg with the new util: https://github.com/micro-ROS/micro-ROS-demos/blob/6cba26276e283b598a5e82c93d365b3709895d0c/rclc/static_type_handling/main.c#L71 
And give feedback about it.

Reply:
Applied on: 9f6efc8f324973cc1538e129697b8ec3774a78ae

Worked as expected. It really simplifies the configuration and allocation of message memory.